flexx/app/_tornadoserver.py

```diff
@@ -678,13 +678,18 @@ def check_origin(self, origin):
         serving_host = self.request.headers.get("Host")
         serving_hostname = serving_host.split(':')[0]
         connecting_host = urlparse(origin).netloc
+        connecting_hostname = connecting_host.split(':')[0]
         
         if serving_hostname == 'localhost':
             return True  # Safe
         elif serving_hostname == '0.0.0.0':
             return True  # we cannot know if the origin matches
         elif serving_host == connecting_host:
             return True
+        elif serving_hostname == connecting_hostname:
+            # Comparing hostnames in case of Jupyter notebooks
+            # connecting_host and serving_host have same host names but different ports
+            return True
```

Comment:
`if manager.has_app_name('__default__'):
            ok_ports = '80', '8888'`
this is ok, but ok_ports should be configurable from file, because if there are several instances of Jupyter server running they are serving different ports, for example 8889, 8890 and etc